app/__init__.py

```diff
@@ -202,7 +202,7 @@ def add_blueprints(application):
 def setup_secure_cookies(application):
     application.secret_key = settings.EQ_SECRET_KEY
     application.permanent_session_lifetime = timedelta(
-        seconds=settings.EQ_SESSION_TIMEOUT_SECONDS)
+        seconds=settings.EQ_SESSION_TIMEOUT_SECONDS + settings.EQ_SESSION_TIMEOUT_GRACE_PERIOD_SECONDS)
```

Comment:
Good spot! It doesn't, and we can't set this via the schema property as this is set on application start for all surveys. The result of this is that the client timeout can be set lower than the server and the client would appear timed out, but the session wouldn't be removed until the server timeout is reached. At the moment this is only used in the test timeout schema, and we should review the implications of this before using it in an actual survey.

At the moment a survey can specify a client timeout greater than the server timeout, which would result in the prompt not being displayed and the user getting a timed out page on the first request after the server timeout. I'll change this so that we always set the client timeout to the same as the server timeout if it's greater than the server timeout.